lib/bloc_record/collection.rb

```diff
@@ -0,0 +1,32 @@
+module BlocRecord
+  class Collection < Array
+    def update_all(updates)
+      ids = self.map(&:id)
+      self.any? ? self.first.class.update(ids, updates) : false
+    end
+
+    def take(num=1)
+      taken_records = Collection.new
+      if num > 1
+        for i in (0..num) do
+          taken_records << self[i]
+        end
+      else
+        taken_records << self.first
```

Comment:
In this case, it should return a _single record_ versus a `Collection` of one record, both per Bloc's `Selection::take` and ActiveRecord's `CollectionProxy::take` implementations. Almost there, though!